src/App.tsx

```diff
@@ -1,24 +1,242 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { FilterOptions } from './types/FilterOptions';
+import { Errors } from './types/Errors';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import * as clientService from './api/todos';
 
-const USER_ID = 0;
+const USER_ID = 44;
+const getFilteredTodos = (todos: Todo[], filterBy: FilterOptions) => {
+  switch (filterBy) {
+    case FilterOptions.Active:
+      return todos.filter(todo => !todo.completed);
+
+    case FilterOptions.Completed:
+      return todos.filter(todo => todo.completed);
+
+    default:
+      return todos;
+  }
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMasage, setErrorMasage] = useState(Errors.noError);
+  const [filterBy, setFilterBy] = useState(FilterOptions.All);
+  const filteredTodos = getFilteredTodos(todos, filterBy);
+  const uncompletedTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todoTitle, setTodoTitle] = useState('');
+  const titleField = useRef<HTMLInputElement>(null);
+  const [isDisabled, setIsDisabled] = useState(false);
```

Comment:
Good practice is firstky define state, then variables
```suggestion
  const [filterBy, setFilterBy] = useState(FilterOptions.All);
  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
  const [todoTitle, setTodoTitle] = useState('');  
  const [isDisabled, setIsDisabled] = useState(false);
  const titleField = useRef<HTMLInputElement>(null);
  
  const filteredTodos = getFilteredTodos(todos, filterBy);
  const uncompletedTodos = todos.filter(todo => !todo.completed);
  const completedTodos = todos.filter(todo => todo.completed);
```